client/stripe-utils/utils.js

```diff
@@ -526,3 +536,34 @@ export const togglePaymentMethodForCountry = ( upeElement ) => {
 		upeContainer.style.display = 'none';
 	}
 };
+
+/**
+ * Unblocks the Block Checkout form.
+ */
+export const unblockBlockCheckout = () => {
+	// Exit early if we're not in a block context.
+	if ( ! wcSettings.wcBlocksConfig ) {
+		return;
+	}
+
+	const checkoutStore = dispatch( CHECKOUT_STORE_KEY );
+
+	// We need to unset the redirect URL otherwise WC core will redirect the the previous checkout redirectURL.
+	// For Wallet payment methods, that will include the #wc-stripe-wallet-... hash and cause the modal to show again.
+	checkoutStore.__internalSetRedirectUrl( null );
+	checkoutStore.__internalSetIdle();
```

Comment:
This branch uses a number of functions marked as internal. I'm not sure if that's not a good idea but this was the only way I could get the payment form to unblock and for the payment elements to be reusable.

Reply:
It works as described, but I agree the internal functions are not ideal... maybe we can get some guidance from @woocommerce/rubik?

Reply:
I'm struggling to understand this, what would cause this flow to trigger? a failed express payment?

Reply:
> I'm struggling to understand this, what would cause this flow to trigger?

So the set up is a little unorthodox. 

When you click the place order button, the process payment request on the server responds with a redirect URL that triggers a modal to be shown.

https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/c3dcabdc-8f8d-4c0f-be6a-7353d4c0d804

When you close that modal, or if payment fails (and Stripe auto close the modal) we needed to reinit the checkout because it was stuck in the "complete" status from the previous request. This code is essentially doing that. 

1) We need to set the URL to `null` because otherwise calling `__internalSetIdle()` would cause WC core to redirect the "currentURL" which was the last URL response. I've lost the copy of the stacktrace I had that indicated this but let me know if you want it I should be able to find it again.
2) I then set the status to idle so the checkout can be resubmitted. Without this the checkout remained completed and the place order button was disabled.